lib/services/bitbucket_commit_hook.rb

```diff
@@ -30,16 +30,21 @@ def receive_webhook
 
   def comment_on_record(ref_num, commit)
     record_type = ref_num =~ /-[0-9]+-/ ? "requirements" : "features"
+    
+    begin
+      email = Mail::Address.new(commit.author.raw)
+    rescue Mail::Field::IncompleteParseError
+      email = commit.author.raw
+    end
 
-    email = Mail::Address.new(commit.author.raw)
-    message = <<-EOF
-      <p>#{email.display_name || email.address} committed:</p>
-      <pre>#{commit.message}</pre>
-      <p>Commit: <a href="#{commit.links.html.href}">#{commit["hash"]}</a></p>
-    EOF
+    message = <<-MESSAGE
+        <p>#{email.try(:display_name) || email.try(:address) || email} committed:</p>
+        <pre>#{commit.message}</pre>
+        <p>Commit: <a href="#{commit.links.html.href}">#{commit['hash']}</a></p>
+    MESSAGE
 
     begin
-      api.create_comment(record_type, ref_num, email.address, message)
+      api.create_comment(record_type, ref_num, email.try(:address), message)
```

Comment:
The possible nil is fine here, the API doesn't actually use the field. I have a sneaking suspicion we can remove the param from the method entirely but I'm a little scared to touch it.

Reply:
Or should we extract `email.try(:display_name) || email.try(:address) || email` from above into a method and use that?

Reply:
The order of priority differs from this line, so I didn't feel it necessary to extract.